### PR TITLE
[parallel, ci] feat: support loading large tensor for fsdp2

### PR DESCRIPTION
**Pull request by JorgenWan**
### What does this PR do?

1. Support loading large tensor for `fsdp2` by `_chunk_and_broadcast_and_dispatch` at the shard dim of `extra parallel`.

The process is as follows:
(1) rank0 loads safetensors from disk to cpu
(2) according to the parallel plan of `extra parallel`, rank0 chunks the loaded tensor at the shard dim
(3) rank0 and other ranks use P2P comm, including `dist.send` and `dist.recv`, to exchange params

This is controlled by `cpu_load_param_name` defined in the `parallel_plan` of specific model. Users can define which parameters are too large that should be loaded from CPU.

Besides, we also provide `max_load_broadcast_size` argument to control the max size of loaded parameters. Parameters larger than this threshold will also use `_chunk_and_broadcast_and_dispatch`.
 
2. Add CI for loading model weights
    1. `load_model_weights`: all ranks load model weights.
    2. `rank0_load_and_broadcast_weights`: rank0 loads and broadcast
    3. The CI uses a toy model with parallel plan of `extra parallel` and `cpu_load_param_name` to test all kind of cases.

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/models/module_utils.py [hunk 1]
```diff
@@ -374,6 +388,175 @@ def _broadcast_and_dispatch(name, shape, dtype, tensor):
...
+                    f"Shard parameter shape = {target_shape}, Loaded parameter shape = {tensor.shape}, pad_size = {pad_size}"
+                )
+
+                tensor = torch.nn.functional.pad(tensor, pad_size, value=0.0)
+
+            assert tensor.shape[0] // para_size == target_shape[0], (
+                f"Parallel {para_group_name}: padded shape {shape} at dim 0 // {para_size} == {tensor.shape[0] // para_size}, not equal to {target_shape[0]}"
+            )
+
+            chunk_loaded_data = list(tensor.chunk(para_size, dim=0))
```
[the remark this answers is not in the document]
Comment (JorgenWan) [reply veomni/models/module_utils.py:490]: fsdp2 uses dim=0 as the default sharding dim

File: docs/usage/arguments.md [hunk 2]
```diff
@@ -292,6 +292,7 @@ without modifying the config class.
 | full_shard | `bool` | `True` | Enable full sharding — equivalent to ZeRO-3. |
 | forward_prefetch | `bool` | `True` | Enable forward prefetch. |
 | offload | `bool` | `False` | Enable CPU offload (FSDP1 only). |
+| max_load_broadcast_size | `float` | `20.0` | Maximum load broadcast size (FSDP2). |
```
Comment (FoolPlayer) [inline docs/usage/arguments.md]: Add unit (GB) for max_load_broadcast_size and detail description for this argument.

**Review by FoolPlayer — APPROVED** [review]
